apps/emqx_connector/src/emqx_connector_mysql.erl

```diff
@@ -106,8 +104,8 @@ on_start(
     Options = [
         {host, Host},
         {port, Port},
-        {user, User},
-        {password, Password},
+        {user, maps:get(username, Config, <<"root">>)},
+        maybe_password_opt(maps:get(password, Config, undefined)),
```

Comment:
1) Maybe we should rather set the default value for `username` in the MySQL connector schema?  That way, I think it would be reflected in the generated documentation.
2) `maybe_password_opt` will return an empty list if the password is `undefined`, so `Options` will be `[{_, _}, ..., [], ... {_, _}]`.  Perhaps the intention was to concat the options with the result of that fn (first wrapping `{password, _}` in a list)?

```erlang
Options = [maybe_password_opt(Password) | ...]
```

Reply:
Sorry I've changed the same code on the master branch, so you have to solve the conflict before merging to master.

Reply:
You're right, the default value should be reflected at the documentation. It's not quite straight-forward to change that though since the definition is shared across multiple schemas. So unfortunately a bit ugly code.

As of `maybe_password_opt`, it's fine to have an empty list as an element in a proplist, even though the spec tells differently. I found, it isn't quite as readable to have the password coming first. That's why I chose the other approach but if this confuses people it's better to use your way then.

Reply:
> Sorry I've changed the same code on the master branch, so you have to solve the conflict before merging to master.

Will deal with it when merging the integration branch.